src/reducer/auth-reducer.js

```diff
@@ -0,0 +1,47 @@
+export const userInitialState = {
+  id: null,
+  firstName: "",
+  lastName: "",
+  isAuthTokenPresent: false,
+  authToken: "",
+  cart: [],
+  wishlist: [],
+  msg: "",
+};
+
+export const authReducer = (state, action) => {
+  switch (action.type) {
+    case "loggedIn": {
+      const { firstName, lastName, _id } = action.payload;
+      const token = action.token;
+      return {
+        ...state,
+        id: _id,
+        firstName: firstName,
+        lastName: lastName,
+        isAuthTokenPresent: true,
+        authToken: token,
+      };
+    }
+    case "signup": {
+      const { firstName, lastName, _id } = action.payload;
+      const token = action.token;
+      return {
+        ...state,
+        id: _id,
+        firstName: firstName,
+        lastName: lastName,
+        isAuthTokenPresent: true,
+        authToken: token,
```

Comment:
You are already assigning the token to `authToken`. So, I think you don't need an extra indicator to check if the token is present or not.

Reply:
okay will look into that thank you :)